web/server.go

```diff
@@ -178,9 +179,26 @@ func NewMux(s *Server) (*http.ServeMux, error) {
 		mux.Handle(stream.DefaultSSEPath, stream.Server(traceAwareSearcher{s.Searcher})) // /stream
 	}
 
+	mux.HandleFunc("/healthz", s.serveHealthz)
+
 	return mux, nil
 }
 
+func (s *Server) serveHealthz(w http.ResponseWriter, r *http.Request) {
+	q := &query.Const{Value: true}
+	opts := &zoekt.SearchOptions{ShardMaxMatchCount: 1, TotalMaxMatchCount: 1, MaxDocDisplayCount: 1}
+
+	result, err := s.Searcher.Search(r.Context(), q, opts)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("not ready: %v", err), http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+
+	_ = json.NewEncoder(w).Encode(result)
```

Comment:
Feels like we should just write `ok` with a `text/plain` as the content type if it works?

Reply:
This felt more debuggable. Any concerns with it?